asr/base.py

```diff
@@ -97,63 +98,66 @@ def review(self):
             estimator=self.model,
             X_training=self.X[init_ind],
             y_training=init_labels,
+            query_strategy=usample_cached,
 
             # additional arguments to pass to fit
             **self.fit_kwargs)
 
         # remove the initial sample from the pool
         pool_ind = np.delete(pool_ind, init_ind)
 
+        self._logger.add_training_log(init_ind, init_labels, i=0)
         query_i = 0
 
         while not self._stop_iter(query_i, pool_ind):
-
+            pred_proba = []
             # Make a query from the pool.
             query_ind, query_instance = self.learner.query(
                 self.X[pool_ind],
-                n_instances=min(self.n_instances, len(pool_ind))  # ,
-                # verbose=self.verbose
+                n_instances=min(self.n_instances, len(pool_ind)),
+                pred_proba=pred_proba,
             )
 
+            # Log the probabilities of samples in the pool being included.
+            if len(pred_proba) > 0:
+                self._logger.add_pool_pred_prob(
```

Comment:
Split this into 2 functions `add_pool_proba` and `add_pool_pred`? More intuitive and flexible.

Reply:
There are two functions: add_pool_pred_prob and add_pool_pred. I don't mind renaming them, but it made sense to me.